src/Mod/Draft/draftutils/make_utils.py

```diff
@@ -0,0 +1,104 @@
+# ***************************************************************************
+# *   (c) 2019 Eliud Cabrera Castillo <e.cabrera-castillo@tum.de>           *
+# *                                                                         *
+# *   This file is part of the FreeCAD CAx development system.              *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   FreeCAD is distributed in the hope that it will be useful,            *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with FreeCAD; if not, write to the Free Software        *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+"""Provides utility functions for make_polarray and make_circulararray."""
+## @package make_utils
+# \ingroup DRAFT
+# \brief This module provides functions for make_polarray & make_circulararray.
+
+import FreeCAD as App
+import Part
+
+import draftutils.utils as utils
+
+from draftutils.messages import _msg, _err
+from draftutils.translate import _tr
+
+
+def make_polcirc_shared(_name, center, axis_object, axis_edge):
```

Comment:
This function is fine but I think it needs to be handled a bit differently.

The `return None` that you see here is meant to exit the `make_polar_array` and `make_circular_array` functions in case of a problem. This means that if you call these functions with wrong arguments, the arguments are checked for correctness, and if they are not correct, the function should terminate, and return a `new_obj = None`.

The way it is done at the moment you call this
```py
axis_reference = make_polcirc_shared(_name, center, axis_object, axis_edge)
```

In case of a problem, `axis_reference = None`, but that's it; the function doesn't exit. For example, if the user inputs a wrong `center`, `axis_reference` is set to `None` but the incorrect `center` is still forwarded to `make_array`, which will probably produce an error.

The way I do it is to collect the correctness status in an additional variable which is returned together with our new `axis_reference` variable. So, we return two things.
```py

    axis_reference = None
    CORRECT = True
    try:
        utils.type_check([(center, App.Vector)], name=_name)
    except TypeError:
        _err(_tr("Wrong input: must be a vector."))
        return not CORRECT, center, axis_reference
    ...
    # Rest of code
```

Then in the `make_*_array` functions we use it like this.
```py
all_correct, center, axis_reference = make_polcirc_shared(_name, center, axis_object, axis_edge)

if not all_correct:
    return None
```

And just a note about the import. I prefer that imported functions from external modules always use the namespace of the module so that we know those functions come from external modules. Something like this.
```py
import draftutils.make_utils as make_utils
...
all_correct, center, axis_reference = make_utils.make_polcirc_shared(...)
```

The only exception to this are the translation functions `translate`, `_tr`, and the messages functions `_msg`, `_wrn`, `_log`, `_err`. This are short and always do the same, so I think it's not necessary to use a namespace for them.

Reply:
So I now worked in that approach. Only variation I did was leaving out returning `center` because it is just being checkerd and not changed.